files/bootstrap.sh

```diff
@@ -455,11 +455,23 @@ if [[ "$USE_MAX_PODS" = "true" ]]; then
   echo "$(jq ".maxPods=$MAX_PODS" $KUBELET_CONFIG)" > $KUBELET_CONFIG
 fi
 
+KUBELET_ARGS="--node-ip=$INTERNAL_IP --pod-infra-container-image=$PAUSE_CONTAINER --v=2"
+
+if vercmp "$KUBELET_VERSION" lt "1.26.0"; then
+  # TODO: remove this when 1.25 is EOL
+  KUBELET_CLOUD_PROVIDER="aws"
+else
+  KUBELET_CLOUD_PROVIDER="external"
+  echo "$(jq ".providerID=\"$(provider-id)\"" $KUBELET_CONFIG)" > $KUBELET_CONFIG
+fi
+
+KUBELET_ARGS="$KUBELET_ARGS --cloud-provider=$KUBELET_CLOUD_PROVIDER"
```

Comment:
Why does this have to stay in the CLI?

Reply:
With kubelet's in-tree cloud providers going away, the docs state that this flag will go away too:
> --cloud-provider
> The provider for cloud services. Set to empty string for running with no cloud provider. If set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used). (DEPRECATED: will be removed in 1.24 or later, in favor of removing cloud provider code from Kubelet.)
> https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/

I'm not clear on kubelet's behavior once this is removed, though. Today, when you use `--cloud-provider=external`:
> kubelets specifying --cloud-provider=external will add a taint `node.cloudprovider.kubernetes.io/uninitialized` with an effect `NoSchedule` during initialization. This marks the node as needing a second initialization from an external controller before it can be scheduled work. Note that in the event that cloud controller manager is not available, new nodes in the cluster will be left unschedulable. The taint is important since the scheduler may require cloud specific information about nodes such as their region or type (high cpu, gpu, high memory, spot instance, etc).
> https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager

If kubelet isn't run on a cloud, it doesn't need to set that label. My assumption is we'll continue to need a boolean `--enable-cloud-provider` flag on kubelet to toggle this behavior.

I plan to bring it up in the next SIG node, maybe the docs are just outdated.

Reply:
Thinking about this more, it's probably more realistic to just deprecate the acceptable values for `--cloud-provider` one-by-one, eventually leaving only `external` and the empty string.

Assuming we ship 1.26 with the external cloud provider, I'd like to get the in-tree `aws` provider removed in 1.28 (assuming it's too late for 1.27)?